### ISSUE TITLE
[BUG] rules with METHOD_CALL locations do not work on calls to static methods

### ISSUE THREAD

**Issue by abrugaro** [konveyor/analyzer-lsp#769]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

latest

### Priority

Undefined (Default)

### Current Behavior

This condition in a rule:

```yaml
  when:
    java.referenced:
      location: METHOD_CALL
      pattern: org.jboss.logging.Logger.getLogger(*)
```
Should trigger an incident in the following code (extracted from [tackle-testapp](https://github.com/konveyor/tackle-testapp-public/blob/main/src/main/java/io/konveyor/demo/ordermanagement/service/CustomerService.java#L20))

```java
private static Logger logger = Logger.getLogger( CustomerService.class.getName() );
```

But it doesn't trigger any incidents.

### How Reproducible

Always (Default)

### Environment

```markdown

```

**Comment by jmle** [comment]
Kind of a uplicate of https://github.com/konveyor/analyzer-lsp/issues/736

**Comment by jmle** [comment]
https://issues.redhat.com/browse/MTA-5763

**Comment by istein1** [comment]
Closing this as it was fixed & verified for v0.8.1
